### ISSUE TITLE
[Task]: Update typehints to support Python 3.11

### ISSUE THREAD

**Issue by AnandInguva** [apache/beam#25591]
### What needs to happen?

To support Python 3.11, we may need to update apache_beam/typehints to resolve any typehint related errors.

Example:  apache_beam/typehints/trivial_inference_test fails with different errors. 
```
self = <apache_beam.typehints.trivial_inference_test.TrivialInferenceTest testMethod=testBuildTupleUnpackWithCall>

    def testBuildTupleUnpackWithCall(self):
      # Lambda uses BUILD_TUPLE_UNPACK_WITH_CALL opcode in Python 3.6, 3.7.
      def fn(x1, x2, *unused_args):
        return x1, x2
    
>     self.assertReturnType(
          typehints.Tuple[typehints.Union[str, float, int],
                          typehints.Union[str, float, int]],
          lambda x1,
          x2,
          _list: fn(x1, x2, *_list), [str, float, typehints.List[int]])

trivial_inference_test.py:372: AssertionError
```

Go through the Python 3.11 change log and modify the typehints code as necessary.

There are so many changes in dis https://docs.python.org/3/library/dis.html, which we use them in https://github.com/apache/beam/blob/master/sdks/python/apache_beam/typehints/trivial_inference.py

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by AnandInguva** [comment]
Some the opcodes I encountered in test failures and added in python 3.11 :

```
'COPY_FREE_VARS'
'RESUME'
'PUSH_NULL'
'POP_JUMP_FORWARD_IF_FALSE'
'BINARY_OP'
'PRECALL'
'MAKE_CELL'
'RETURN_GENERATOR'
```

**Comment by jrmccluskey** [comment]
This was resolved by #25798 and #25795, just never got linked to this issue
